bee-tangle/src/traversal.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Collection of Tangle traversal functions.
+
+// TODO: Refactor all of this into methods on `Tangle`.
+
+use crate::{
+    tangle::{Hooks, Tangle},
+    MessageRef,
+};
+
+use bee_message::MessageId;
+
+use std::{collections::HashSet, future::Future};
+
+// /// A Tangle walker that - given a starting vertex - visits all of its ancestors that are connected through
```

Comment:
Old code?

Reply:
@thibault-martinez Do you want this code kept?

Reply:
No